packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -480,24 +481,26 @@ Future<void> _buildGradleProjectV2(
   }
 
   if (!isBuildingBundle) {
-    final File apkFile = _findApkFile(project, buildInfo);
-    if (apkFile == null)
+    final Iterable<File> apkFiles = _findApkFiles(project, buildInfo);
+    if (apkFiles.isEmpty)
       throwToolExit('Gradle build failed to produce an Android package.');
-    // Copy the APK to app.apk, so `flutter run`, `flutter install`, etc. can find it.
-    apkFile.copySync(project.apkDirectory.childFile('app.apk').path);
+    // Copy the first APK to app.apk, so `flutter run`, `flutter install`, etc. can find it.
+    apkFiles.first.copySync(project.apkDirectory.childFile('app.apk').path);
```

Comment:
This might choose an APK that doesn't match the device's arch, no?  Seems like we probably need to refactor our logic that currently looks in `project.apkDirectory.childFile('app.apk')` to handle multiple APKs and choose the right one when installing, running, etc.

(beyond the scope of this change though - can be a TODO for now)

Reply:
Added the TODO.